### PR TITLE
Feat/node client

### PR DESCRIPTION
**Pull request by Frenchcooc**
# Description

Provide a Node.js client for Pizzly. This make it easier to perform requests to third-party API through Pizzly proxy.

# Example

```js
const Pizzly = require('pizzly')

const pizzly = new Pizzly({ host: 'pizzly.example.org' }) // Initialize Pizzly with your own instance
const myAPI = pizzly.integration('x-api') // Replace with the API slugname

myAPI
  .auth('x-auth-id') // Replace with a valid authId
  .get('/x-endpoint') // Replace with a valid endpoint
  .then(response => console.log) // Do something with the response
  .catch(console.error)
```

# Under the hood

Most code is shared with the JS client, but not everything. So it makes sense to have two separate folders/projects. 
This client uses the `node-fetch` package ([link](https://github.com/node-fetch/node-fetch/)) to perform API requests.

# Next steps

Once approved, this will be published under NPM to ease the installation:

```bash
npm install pizzly-node
```

### CODE REVIEW

File: src/clients/node/README.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+```
+
+### Passing extra params
+
+Here's how to send headers to the API:
+
+```js
+myAPI
+  .auth('x-auth-id')
+  .post('/x-endpoint', { headers: { 'Content-Type': 'multipart/form-data; boundary=something' } })
```
Comment (nabilfreeman) [inline src/clients/node/README.md:69]: 👏👏👏

**Comment by nabilfreeman** [comment]
Nice job! Thanks for turning this around so fast. I looked through the differences between the JS client and Node client, namely `integration.ts`, and I can see that they are different enough that it's not really necessary to have class inheritance or anything to share the code.

Not much to complain about as I think the original `.integration().auth()...` flow is very elegant.

What else could be added to the enclosing `headers` object? Could anything compatible with Fetch request options be passed here? See *"Supplying request options"* https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch

**Comment by Frenchcooc** [comment]
Hi Nabil, thanks for the feedback! I believe we could easily add more properties when the case needed. Having `headers`, `body` and `query` seems good enough for now ;)

**Comment by nabilfreeman** [comment]
👏

**Pull request merged**